tests/cppunit/disk_test.cc

```diff
@@ -0,0 +1,202 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+#include <chrono>
+#include <gtest/gtest.h>
+#include <memory>
+#include <vector>
+#include <thread>
+#include "redis_metadata.h"
+#include "test_base.h"
+#include "redis_string.h"
+#include "redis_disk.h"
+#include "redis_set.h"
+#include "redis_list.h"
+#include "redis_zset.h"
+#include "redis_bitmap.h"
+#include "redis_sortedint.h"
+
+class RedisDiskTest : public TestBase {
+protected:
+  explicit RedisDiskTest() : TestBase() {
+  }
+  ~RedisDiskTest() = default;
+
+protected:
+};
+
+TEST_F(RedisDiskTest, StringDisk) {
+  key_ = "stringdisk_key";
+  std::unique_ptr<Redis::String> string = Util::MakeUnique<Redis::String>(storage_, "disk_ns_string");
+  std::unique_ptr<Redis::Disk> disk = Util::MakeUnique<Redis::Disk>(storage_, "disk_ns_string");
+  std::vector<int> value_size{1, 1024};
+  for(auto &p : value_size){
+    EXPECT_TRUE(string->Set(key_, std::string(p, 'a')).ok());
+    std::string got_value;
+    EXPECT_TRUE(string->Get(key_,  &got_value).ok());
+    EXPECT_EQ(got_value, std::string(p, 'a'));
+    std::string value;
+    string->Get(key_, &value);
+    uint64_t result = 0;
+    EXPECT_TRUE(disk->GetKeySize(key_, kRedisString, &result).ok());
+    EXPECT_GE(result, p);
+    break;
```

Comment:
I'm wondering why this operator (`break`) is here. With it, the loop will run only one time. But if you want to do something just one time, there is no reason to introduce the loop.

Reply:
> I'm wondering why this operator (`break`) is here. With it, the loop will run only one time. But if you want to do something just one time, there is no reason to introduce the loop.

I noticed that the running results of ci are somewhat different from those of my native machine. At the same time, because this is a draft pr, I am using ci for testing.I found that a byte of data is often not detected due to the problem with the settings of Writebatch(sync is default false). So don't worry. I'll fix it after the test. Thanks again for your serious review